pkg/csi/cinder/openstack/openstack_backups.go

```diff
@@ -0,0 +1,195 @@
+	backupReadyDuration                  = 5 * time.Second
+	backupReadyFactor                    = 1.2
+	backupReadySteps                     = 15
+	BackupMaxDurationSecondsPerGBDefault = 20
+	BackupMaxDurationPerGB               = "backup-max-duration-seconds-per-gb"
+)
+
+// CreateBackup issues a request to create a Backup from the specified Snapshot with the corresponding ID and
+// returns the resultant gophercloud Backup Item upon success.
+func (os *OpenStack) CreateBackup(name, volID string, snapshotID string, tags *map[string]string) (*backups.Backup, error) {
+	blockstorageServiceClient, err := openstack.NewBlockStorageV3(os.blockstorage.ProviderClient, os.epOpts)
+	if err != nil {
+		return &backups.Backup{}, err
+	}
+
+	force := false
+	// if no flag given, then force will be false by default
+	// if flag it given , check it
+	if item, ok := (*tags)[SnapshotForceCreate]; ok {
+		var err error
+		force, err = strconv.ParseBool(item)
+		if err != nil {
+			klog.V(5).Infof("Make force create flag to false due to: %v", err)
+		}
+		delete(*tags, SnapshotForceCreate)
+	}
+
+	opts := &backups.CreateOpts{
+		VolumeID:    volID,
+		SnapshotID:  snapshotID,
+		Name:        name,
+		Force:       force,
+		Description: backupDescription,
+	}
+
+	if tags != nil {
+		// Set openstack microversion to 3.43 to send metadata along with the backup
+		blockstorageServiceClient.Microversion = "3.43"
+		opts.Metadata = *tags
+	}
+
+	// TODO: Do some check before really call openstack API on the input
+	mc := metrics.NewMetricContext("backup", "create")
+	backup, err := backups.Create(blockstorageServiceClient, opts).Extract()
+	if mc.ObserveRequest(err) != nil {
+		return &backups.Backup{}, err
+	}
+	// There's little value in rewrapping these gophercloud types into yet another abstraction/type, instead just
+	// return the gophercloud item
+	return backup, nil
+}
+
+// ListBackups retrieves a list of active backups from Cinder for the corresponding Tenant.  We also
+// provide the ability to provide limit and offset to enable the consumer to provide accurate pagination.
+// In addition the filters argument provides a mechanism for passing in valid filter strings to the list
+// operation.  Valid filter keys are:  Name, Status, VolumeID, Limit, Marker (TenantID has no effect).
+func (os *OpenStack) ListBackups(filters map[string]string) ([]backups.Backup, error) {
+	var allBackups []backups.Backup
+
+	// Build the Opts
+	opts := backups.ListOpts{}
+	for key, val := range filters {
+		switch key {
+		case "Status":
+			opts.Status = val
+		case "Name":
+			opts.Name = val
+		case "VolumeID":
+			opts.VolumeID = val
+		case "Marker":
+			opts.Marker = val
+		case "Limit":
+			opts.Limit, _ = strconv.Atoi(val)
+		default:
+			klog.V(3).Infof("Not a valid filter key %s", key)
+		}
+	}
+	mc := metrics.NewMetricContext("backup", "list")
+
+	allPages, err := backups.List(os.blockstorage, opts).AllPages()
+	if err != nil {
+		panic(err)
```

Comment:
why do you use `panic(err)`?

```suggestion
		return nil, err
```